packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,528 @@
+            },
+          },
+        },
+      },
+    });
+
+    const childTemplate = parent.getNestedStack('ChildStack').includedTemplate;
+    const grandChild = childTemplate.getNestedStack('GrandChildStack');
+    const bucket = grandChild.includedTemplate.getResource('BucketImport') as s3.CfnBucket;
+
+    bucket.bucketName = 'modified-bucket-name';
+
+    expect(grandChild.stack).toHaveResource('AWS::S3::Bucket', { BucketName: 'modified-bucket-name' });
+  });
+
+  test('asset parameters generated in parent and child are identical', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-onechild.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+        },
+      },
+    });
+
+    const assetParam = 'AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0C';
+    const assetParamKey = 'AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2';
+    expect(stack).toMatchTemplate({
+      "Parameters": {
+        [assetParam]: {
+          "Type": "String",
+          "Description": "S3 bucket for asset \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\"",
+        },
+        [assetParamKey]: {
+          "Type": "String",
+          "Description": "S3 key for asset version \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\"",
+        },
+        "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50ArtifactHash9C417847": {
+          "Type": "String",
+          "Description": "Artifact hash for asset \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\"",
+        },
+      },
+      "Resources": {
+        "ChildStack": {
+          "Type": "AWS::CloudFormation::Stack",
+          "Properties": {
+            "TemplateURL": {
+              "Fn::Join": [ "", [
+                "https://s3.",
+                { "Ref": "AWS::Region" },
+                ".",
+                { "Ref": "AWS::URLSuffix" },
+                "/",
+                { "Ref": assetParam },
+                "/",
+                { "Fn::Select": [
+                  0,
+                  { "Fn::Split": [
+                    "||",
+                    { "Ref": assetParamKey },
+                  ]},
+                ]},
+                { "Fn::Select": [
+                  1,
+                  { "Fn::Split": [
+                    "||",
+                    { "Ref": assetParamKey },
+                  ]},
+                ]},
+              ]],
+            },
+            "Parameters": {
+              "MyBucketParameter": "some-magic-bucket-name",
+            },
+          },
+        },
+      },
+    });
+  });
+
+  test('templates with nested stacks that do not exist in the nested stacks property have those stacks ignored', () => {
```

Comment:
This phrasing is weird... they are not ignored, right? Otherwise they would disappear from the template?

```suggestion
  test('templates with nested stacks that were not provided in the nestedStacks property are left unmodified', () => {
```